src/manifolds/Hyperbolic.jl

```diff
@@ -380,3 +380,22 @@ for (P, T) in zip(_ExtraHyperbolicPointTypes, _ExtraHyperbolicTangentTypes)
     @eval zero_vector(::Hyperbolic, p::$P) = $T(zero(p.value))
     @eval zero_vector!(::Hyperbolic, X::$T, ::$P) = fill!(X.value, 0)
 end
+
+@doc raw"""
+    riemann_tensor(M::Hyperbolic{n}, p, X, Y, Z)
+
+Compute the Riemann tensor ``R(X,Y)Z`` at point `p` on [`Hyperbolic`](@ref) `M`.
+The formula reads (see e.g., [Lee:2019](@cite) Proposition 8.36)
+
+````math
+R(X,Y)Z = - (\langle Z, Y \rangle X - \langle Z, X \rangle Y)
+````
+"""
+riemann_tensor(::Hyperbolic, p, X, Y, Z)
+
+function riemann_tensor!(M::Hyperbolic, Xresult, p, X, Y, Z)
+    innerZX = inner(M, p, Z, X)
+    innerZY = inner(M, p, Z, Y)
+    Xresult .= -(innerZY .* X .- innerZX .* Y)
```

Comment:
```suggestion
    Xresult .= innerZX .* Y .- innerZY .* X
```
This should reduce the number of operations required here.